src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerCodeBlocks.kt

```diff
@@ -0,0 +1,68 @@
+package com.cjbooms.fabrikt.generators.controller
+
+import com.cjbooms.fabrikt.generators.controller.metadata.SpringImports
+import com.cjbooms.fabrikt.model.IncomingParameter
+import com.squareup.kotlinpoet.CodeBlock
+
+data class SimpleCodeBlock(val body: String, val args: List<Any> = emptyList())
+
+fun SimpleCodeBlock.toCodeBlock() = CodeBlock.of(body, *args.toTypedArray())
+
+object ControllerCodeBlocks {
+
+    fun postWithResponsebody(serviceFuncName: String, parameters: List<IncomingParameter>): SimpleCodeBlock {
+        return SimpleCodeBlock(
+            """return ResponseEntity
+                .ok(${serviceCallLine(serviceFuncName, parameters)})
+            """
+        )
+    }
+
+    fun postWithLocationResponse(serviceFuncName: String, parameters: List<IncomingParameter>): SimpleCodeBlock {
+        return SimpleCodeBlock(
+            """val svcResp = ${serviceCallLine(serviceFuncName, parameters)}
+            val uri = %T.fromCurrentRequest().path("/${"$"}{svcResp.id}").build().toUri()
+            return ResponseEntity.created(uri).build()
```

Comment:
Possibly more logic that should be pushed into the service layer. We are making assumptions about how people want to create their resource URIs

Reply:
Fully agreed. I guess the problem we have here is that the URI scope lays at the resource layer. How do we resolve this?

1)  Return a compose object from the service with `uri` and response object?
  ```
   data class LocationResponse<T>(T: responseObject, URI: uri)
```
2) Remove the postWithLocationResponse option in its entirety and let the user the option to construct it if he needs to?
3) Build a stub instead
4) Any other ?

Reply:
I'd go for 1.